store/store.go

```diff
@@ -135,9 +150,25 @@ func (s *Store) reregisterSchemas() error {
 
 	for res := range results.Next() {
 		name := ds.RawKey(res.Key).Name()
-		if _, err = s.RegisterSchema(name, string(res.Value)); err != nil {
+		m, err := s.RegisterSchema(name, string(res.Value), nil)
+		if err != nil {
 			return err
 		}
+		indexes, err := s.datastore.Query(query.Query{
+			Prefix:   indexPrefix.Child(m.BaseKey()).String(),
+			KeysOnly: true,
+		})
+		if err != nil {
+			return err
+		}
+		defer indexes.Close()
+		for index := range indexes.Next() {
+			path := ds.RawKey(index.Key).Name()
+			if err := m.AddIndex(path, false); err != nil { // @todo: Enable unique constraint
```

Comment:
For discussion: We could encode the unique constraint into the index value for easy rehydration, or we could expect callers to specify the index each time, in which case, we wouldn't even really need this hydration step.